### PR TITLE
postgresql: enable pie

### PR DESCRIPTION
**Pull request by Mic92**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Added a release notes entry if the change is major or breaking
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 124188` at 0ba0fda6 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>576 packages marked as broken and skipped:</summary>
  <ul>
    <li>aqemu</li>
    <li>bareos</li>
    <li>clickshare-csc1</li>
    <li>gnuradio3_7</li>
    <li>gnuradio3_7Packages.ais</li>
    <li>gnuradio3_7Packages.gnuradio</li>
    <li>gnuradio3_7Packages.gsm</li>
    <li>gnuradio3_7Packages.limesdr</li>
    <li>gnuradio3_7Packages.nacl</li>
    <li>gnuradio3_7Packages.osmosdr</li>
    <li>...</li>
  </ul>
</details>
<details>
  <summary>1930 packages skipped due to time constraints:</summary>
  <ul>
    <li>AusweisApp2</li>
    <li>OSCAR</li>
    <li>acoustidFingerprinter</li>
    <li>adminer</li>
    <li>adwaita-qt</li>
    <li>aj-snapshot</li>
    <li>ajour</li>
    <li>albert</li>
    <li>aliza</li>
    <li>amarok (amarok-kf5)</li>
    <li>...</li>
  </ul>
</details>
<details>
  <summary>10 packages built successfully:</summary>
  <ul>
    <li>alerta-server</li>
    <li>inspircd</li>
    <li>movine</li>
    <li>postgresql (postgresql_11)</li>
    <li>postgresqlPackages.pgvector (postgresql11Packages.pgvector)</li>
    <li>postgresql_10</li>
    <li>python38Packages.psycopg2</li>
    <li>rubyPackages_3_0.pg</li>
    <li>rubyPackages_3_0.sequel_pg</li>
    <li>teamspeak_server</li>
  </ul>
</details>
<details>
  <summary>2 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/stale-substitute.md">stale-substitute</a>

Stale substituteInPlace detected.
substituteStream(): WARNING: pattern '/nix/store/35pnk5kwi26m3ph2bc7dxwjnavpzl8cn-gcc-wrapper-10.3.0/bin/ld' doesn't match anything in file '/nix/store/gg0g4dd4swpcc78gg5nhb7l39h525c5z-postgresql-11.11/lib/pgxs/src/Makefile.global'
Near pkgs/servers/sql/postgresql/default.nix:160:0:
```
    |
160 |       description = "A powerful, open source object-relational database system";
    | ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/stale-substitute.md">stale-substitute</a>

Stale substituteInPlace detected.
substituteStream(): WARNING: pattern '/nix/store/35pnk5kwi26m3ph2bc7dxwjnavpzl8cn-gcc-wrapper-10.3.0/bin/ld' doesn't match anything in file '/nix/store/gh953ba0kbmhhvas8zzb6rwiw61vxy5w-postgresql-10.16/lib/pgxs/src/Makefile.global'
Near pkgs/servers/sql/postgresql/default.nix:160:0:
```
    |
160 |       description = "A powerful, open source object-relational database system";
    | ^
```
</li>
  </ul>
</details>

File: pkgs/servers/sql/postgresql/default.nix [hunk 1]
```diff
@@ -33,6 +33,8 @@ let
       inherit sha256;
     };
 
+    hardeningEnable = [ "pie" ];
```
Comment (marsam) [inline pkgs/servers/sql/postgresql/default.nix:36]: it fails on Darwin
```suggestion
    hardeningEnable = lib.optionals (!stdenv.isDarwin) [ "pie" ];
```

btw, this PR should target staging

Comment (mohe2015) [reply pkgs/servers/sql/postgresql/default.nix:36]: Is the condition correct or do you need to  invert it?

Comment (Mic92) [reply pkgs/servers/sql/postgresql/default.nix]: Is `pie` not support on darwin by default?

Comment (ajs124) [reply pkgs/servers/sql/postgresql/default.nix]: AFAICT this isn't a darwin but an LLVM problem. Trying to build postgres on Linux with LLVM fails during configuring like this:
```
checking for __builtin_bswap16... yes
checking for __builtin_bswap32... yes
checking for __builtin_bswap64... yes
checking for __builtin_clz... yes
checking for __builtin_ctz... yes
checking for __builtin_popcount... yes
checking for _LARGEFILE_SOURCE value needed for large files... no
checking how cc reports undeclared, standard C functions... configure: error: in `/build/postgresql-13.3':
configure: error: cannot detect from compiler exit status or warnings
See `config.log' for more details
builder for '/nix/store/bg5yp1lc6kfyd0pnvmb369a2q9mh3wvx-postgresql-13.3.drv' failed with exit code 1
```

Overriding `hardeningEnable` to `[]` makes it build again.

**Pull request merged**